hail/python/hail/matrixtable.py

```diff
@@ -2554,13 +2557,11 @@ def _annotate_all(self,
         base, cleanup = self._process_joins(*all_exprs)
         jmt = base._jvds
         if row_exprs:
-            row_key = None if len(set(self.row_key) & set(row_exprs.keys())) == 0 else self.row_key
             row_struct = InsertFields(base.row._ir, [(n, e._ir) for (n, e) in row_exprs.items()])
-            jmt = jmt.selectRows(str(row_struct), row_key)
+            jmt = jmt.selectRows(str(row_struct))
         if col_exprs:
-            col_key = None if len(set(self.col_key) & set(col_exprs.keys())) == 0 else self.col_key
             col_struct = InsertFields(base.col._ir, [(n, e._ir) for (n, e) in col_exprs.items()])
-            jmt = jmt.selectCols(str(col_struct), col_key)
+            jmt = jmt.selectCols(str(col_struct), None)
```

Comment:
Why doesn't `selectCols` match `selectRows`?

Reply:
Because `selectCols` is still the combined map/keyBy for columns. I changed `_annotate_all` to have the same semantics as all similar operations in python: key fields must not be modified (which was already the case for all callers). The `None` argument to `selectCols` expresses that we aren't changing the column keys.